chemqulacs/vqe/vqeci.py

```diff
@@ -437,24 +451,46 @@ def kernel(self, h1, h2, norb, nelec, ecore=0, **kwargs):
             self.singlet_excitation,
         )
         # Create parametric state
-        param_circuit = LinearMappedUnboundParametricQuantumCircuit(self.n_qubit)
-        param_circuit.extend(self.initial_state.circuit)
-        param_circuit.extend(ansatz)
-        param_state = ParametricCircuitQuantumState(self.n_qubit, param_circuit)
+        param_states = []
+        for i in range(len(self.initial_states)):
+            param_circuit = LinearMappedUnboundParametricQuantumCircuit(self.n_qubit)
+            param_circuit.extend(self.initial_states[i].circuit)
+            param_circuit.extend(ansatz)
+            param_state = ParametricCircuitQuantumState(self.n_qubit, param_circuit)
+            param_states.append(param_state)
 
         gradient_estimator = create_parameter_shift_gradient_estimator(
             self.parametric_estimator
         )
 
+        def get_energies(params):
+            return [
+                self.parametric_estimator(qubit_hamiltonian, param_states[i], [params])[
+                    0
+                ].value.real
+                for i in range(len(param_states))
+            ]
+
         def cost_fn(params):
-            estimate = self.parametric_estimator(
-                qubit_hamiltonian, param_state, [params]
-            )[0]
-            return estimate.value.real
+            return sum(
+                [
+                    self.parametric_estimator(
+                        qubit_hamiltonian, param_states[i], [params]
+                    )[0].value.real
+                    * 2 ** (-i)
```

Comment:
Add "weight_policy" option on VQECASCI and VQECASSCF, so that we can choose the weights we use for SSVQE.

We can chose from 3 options.
same : weight = 1 , 1 , 1 , 1 , ...
base_first : weight = 1, 0.5 , 0.5 , 0.5 ...
exponential : weight = 1, 0.5 , 0.25 , 0.125 ...